teku/src/main/java/tech/pegasys/teku/cli/subcommand/ValidatorClientCommand.java

```diff
@@ -94,9 +111,40 @@ public Integer call() {
     return 1;
   }
 
+  private void configureWithSpecFromBeaconNode(Eth2NetworkConfiguration.Builder builder) {
+    try {
+      var spec = getSpecWithRetry(validatorClientOptions.parseApiEndpoint(), 4);
```

Comment:
We don't want to limit the number of retries here - the validator client should just keep retrying until the beacon node becomes available.

Reply:
Ok, could there be legitimate scenarios where a beacon node never comes online though and this process endlessly retries? Could this be a problem in certain setups? In any case, I agree that fixing an arbitrary deadline here isn't ideal either. Preferably, this would be something user-specified through an option, but that might be overcomplicating a simple capability. I'll remove retry limits for now...

Reply:
It's certainly possible the user might specify the wrong URL for the beacon node and it would never work but it's far more likely that the beacon node is either still starting up or undergoing maintenance. Having it constantly retry means you no longer need to worry about the order of startup - the validator client can start up first and will just start working whenever the beacon node is up.

Given how early in the process we're retrieving the spec, even the metrics endpoint won't have started so any kind of monitoring will flag that the validator client isn't working (and the logs/console will be filled with errors).